astropy/visualization/wcsaxes/patches.py

```diff
@@ -81,9 +84,20 @@ class SphericalCircle(Polygon):
 
     def __init__(self, center, radius, resolution=100, vertex_unit=u.degree, **kwargs):
 
-        # Extract longitude/latitude, either from a tuple of two quantities, or
-        # a single 2-element Quantity.
-        longitude, latitude = center
+        # Extract longitude/latitude, either from a SkyCoord object, or
+        # from a tuple of two quantities or a single 2-element Quantity.
+        # The SkyCoord is converted to SphericalRepresentation, if not already.
+        if isinstance(center, SkyCoord):
+            rep_type = center.representation_type
+            if rep_type != SphericalRepresentation:
+                warnings.warn(f'Received center of representation type {rep_type} '
+                              'will be converted to Spherical Representation ',
+                              UserWarning)
+                center.representation_type = 'spherical'
+            attr_names = list(center.representation_component_names.keys())
```

Comment:
Converting a dictionary into a list in Python only converts the keys, so 
```suggestion
            attr_names = list(center.representation_component_names)
```
would work equally well.